src/main/java/com/aws/greengrass/secretmanager/LocalStoreMap.java

```diff
@@ -0,0 +1,303 @@
+package com.aws.greengrass.secretmanager;
+
+import com.aws.greengrass.config.Node;
+import com.aws.greengrass.deployment.DeviceConfiguration;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.secretmanager.crypto.Crypter;
+import com.aws.greengrass.secretmanager.crypto.KeyChain;
+import com.aws.greengrass.secretmanager.crypto.MasterKey;
+import com.aws.greengrass.secretmanager.crypto.RSAMasterKey;
+import com.aws.greengrass.secretmanager.exception.SecretCryptoException;
+import com.aws.greengrass.secretmanager.exception.SecretManagerException;
+import com.aws.greengrass.secretmanager.model.AWSSecretResponse;
+import com.aws.greengrass.secretmanager.model.SecretConfiguration;
+import com.aws.greengrass.secretmanager.model.SecretDocument;
+import com.aws.greengrass.secretmanager.store.FileSecretStore;
+import com.aws.greengrass.secretmanager.store.SecretStore;
+import com.aws.greengrass.security.SecurityService;
+import com.aws.greengrass.security.exceptions.ServiceUnavailableException;
+import com.aws.greengrass.util.Coerce;
+import com.aws.greengrass.util.LockFactory;
+import com.aws.greengrass.util.LockScope;
+import com.aws.greengrass.util.RetryUtils;
+import com.aws.greengrass.util.Utils;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
+
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyPair;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.Lock;
+import javax.inject.Inject;
+
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_CERTIFICATE_FILE_PATH;
+import static com.aws.greengrass.deployment.DeviceConfiguration.DEVICE_PARAM_PRIVATE_KEY_PATH;
+
+public class LocalStoreMap {
+    private final Logger logger = LogManager.getLogger(LocalStoreMap.class);
+    private final ConcurrentHashMap<String, Labels> secrets;
+    private final SecretStore<SecretDocument, AWSSecretResponse> secretStore;
```

Comment:
Since `secretStore` is now in `LocalStoreMap.java`, can we remove the `secretStore` instance in `SecretManager.java`?

Reply:
It is used by the cache as well.